### PR TITLE
Create "Add dynamic entity layer" Sample

### PR DESCRIPTION
**Pull request by amweis226**
# Description

Adds new "Add dynamic entity layer" sample to the sample viewer.

## Type of change

- New sample implementation

## Platforms tested on

- [x] WPF .NET 6
- [x] WPF Framework
- [x] WinUI
- [x] MAUI WinUI
- [x] MAUI Android
- [x] MAUI iOS
- [x] MAUI MacCatalyst

## Checklist

- [x] Self-review of changes
- [x] All changes work as expected on all affected platforms
- [x] There are no warnings related to changes
- [x] Code is commented and follows .NET conventions and standards
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] No unrelated changes have been made to any other code or project files
- [x] Screenshots are correct size and display in description tab (800 x 600 on Windows, 600 height mobile screenshots for MAUI)

### CODE REVIEW

**Comment by amweis226** [comment]
There is issue with the label for previous observations slider in MAUI. This platform's version doesn't have tick and frequency controls like WinUI and WPF so snapping values to an integer is not possible directly on the label control. This does not impact the functionality of the app. Any suggestions are welcome.

File: src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                <Grid>
+                    <Grid.ColumnDefinitions>
+                        <ColumnDefinition Width="auto" />
+                        <ColumnDefinition Width="auto" />
+                    </Grid.ColumnDefinitions>
+                    <Grid.RowDefinitions>
+                        <RowDefinition Height="auto" />
+                        <RowDefinition Height="auto" />
+                    </Grid.RowDefinitions>
+                    <Label Grid.Row="0"
```
Comment (williambohrmann3) [inline src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml:26]: For this entire grid, I'd swap columns for each control, so the checkboxes in column 0 and labels in column 1. Adding VerticalOptions="Center" to both labels will align the label next to the checkbox.

File: src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        <Border x:Name="MainUI"
+                Style="{DynamicResource EsriSampleControlPanel}"
+                VerticalOptions="Start">
+            <StackLayout>
+                <Label x:Name="ConnectionStatusLabel" />
+                <Button x:Name="ConnectionButton"
+                        Margin="5"
+                        Clicked="ConnectionButtonClick"
+                        IsEnabled="False" />
+                <Grid>
```
Comment (williambohrmann3) [inline src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml:17]:
```suggestion
                <Grid ColumnDefinitions="auto, auto" RowDefinitions="auto, auto">
```
Shorthand for column/row definitions.

File: src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                           Grid.Column="0"
+                           Text="Previous observations" />
+                    <CheckBox Grid.Row="1"
+                              Grid.Column="1"
+                              Margin="5"
+                              CheckedChanged="EntityVisibilityCheckbox_Checked"
+                              IsChecked="True" />
+                </Grid>
+                <Label BindingContext="{x:Reference EntitiesSlider}"
+                       Text="{Binding Value, StringFormat='Observations per track ({0})'}"
```
Comment (williambohrmann3) [inline src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml:44]: I'd set the text of this label in the code behind, in your Slider_ValueChanged event handler. That way you're able to display EntitiesSlider.Value as a whole number.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                          IsChecked="True"
+                          Unchecked="EntityVisibilityCheckbox_Checked" />
+                <Grid>
+                    <Grid.ColumnDefinitions>
+                        <ColumnDefinition Width="0.2*" />
+                        <ColumnDefinition Width="*" />
+                    </Grid.ColumnDefinitions>
+                    <TextBlock Width="155"
+                               VerticalAlignment="Center"
+                               Text="{Binding ElementName=EntitiesSlider, Path=Value}" />
```
Comment (williambohrmann3) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml:32]: For clarity, "Observations per track" should preface EntitiesSlider.Value

**Review by williambohrmann3 — CHANGES_REQUESTED** [review]
Functional, looks great, just needs a few final UI tweaks.

**Review by duffh — CHANGES_REQUESTED** [review]
MAUI and WPF looks great.

WinUI might be a little large in terms of UI.

Couldn't test MAUI iOS or MacCatalyst without 200.1 packages. 

I'd recommend merging in `v.next` to update the screenshots.

Your checks are failing, once everything has been addressed I'll take a look.

File: src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                              Grid.Column="0"
+                              Margin="5"
+                              CheckedChanged="EntityVisibilityCheckbox_Checked"
+                              IsChecked="True" />
+                    <Label Grid.Row="1"
+                           Grid.Column="1"
+                           Text="Previous observations"
+                           VerticalTextAlignment="Center" />
+                </Grid>
+                <Label x:Name="previousObservationsLabel" Text="Previous observations per track: 5" />
```
Comment (williambohrmann3) [inline src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml:45]: When observations number changes from one to two digits (and vice versa), the MapView will subtly shrink/expand. This is a landscape orientation issue specific to mobile (noticed this on OnePlus Android device).

File: src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage x:Class="ArcGIS.Samples.AddDynamicEntityLayer.AddDynamicEntityLayer"
+             xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:esriUI="clr-namespace:Esri.ArcGISRuntime.Maui;assembly=Esri.ArcGISRuntime.Maui">
+    <Grid Style="{DynamicResource EsriSampleContainer}">
+        <esriUI:MapView x:Name="MyMapView" Style="{DynamicResource EsriSampleGeoView}" />
+        <Border x:Name="MainUI"
+                Style="{DynamicResource EsriSampleControlPanel}"
```
Comment (williambohrmann3) [inline src/MAUI/Maui.Samples/Samples/Layers/AddDynamicEntityLayer/AddDynamicEntityLayer.xaml:9]: On iPhone Simulator (SE 3 and 14 Pro Max), if you start the sample in vertical mode, then rotate to horizontal mode, the control panel takes up the entire screen. If you start sample in horizontal mode, it'll display the control panel and map just fine. However, upon rotating the screen to vertical mode and back to horizontal mode, the map view is invisible.

**Review by williambohrmann3 — CHANGES_REQUESTED** [review]
Desktop looks great. Two mobile issues.

**Review by duffh — APPROVED** [review]
Approved pending resolution of William's comments

**Review by williambohrmann3 — APPROVED** [review]

**Pull request merged**